packages/runtime/src/widgets/Dropdown.tsx

```diff
@@ -1,34 +1,59 @@
 import { Form as AntForm, Select } from "antd";
-import { useCallback, useState } from "react";
-import { useRegisterBindings } from "@ensembleui/react-framework";
-import type { EnsembleAction, Expression } from "@ensembleui/react-framework";
+import { useCallback, useMemo, useState } from "react";
+import {
+  CustomScopeProvider,
+  evaluate,
+  useRegisterBindings,
+  useTemplateData,
+  defaultScreenContext,
+  unwrapWidget,
+} from "@ensembleui/react-framework";
+import type {
+  CustomScope,
+  EnsembleAction,
+  EnsembleWidget,
+  Expression,
+  TemplateData,
+} from "@ensembleui/react-framework";
+import { get, isObject, isString } from "lodash-es";
 import { WidgetRegistry } from "../registry";
 import type { EnsembleWidgetProps } from "../shared/types";
 import { useEnsembleAction } from "../runtime/hooks/useEnsembleAction";
+import { EnsembleRuntime } from "../runtime";
+
+export interface DropdownItemTemplace {
+  data: Expression<TemplateData>;
+  name: string;
+  template: EnsembleWidget;
+  value: Expression<string>;
+}
 
 export type DropdownProps = {
   label?: string;
   hintText?: string;
   value?: Expression<string | number>;
-  items: SelectOption[];
+  items?: SelectOption[];
   onItemSelect: EnsembleAction;
   autoComplete: Expression<boolean>;
+  "item-template"?: DropdownItemTemplace;
```

Comment:
There is a shared type for this

Reply:
Yup, I know. but here we need extra args inside the `item-template`. so I used it like this

Reply:
You can use type composition: `HasItemTemplate & { value: Expression<string> }`